### ISSUE TITLE
1.21?

### ISSUE THREAD

**Issue by MolodoyDeputat** [fxmorin/carpet-fixes#178]

**Comment by lighthal** [comment]
It would be nice to have the carpet duplication glitch fixed in 1.21.

**Comment by DustFallee** [comment]
我非常需要它的“基岩版飞行”功能

**Comment by alien-sunset** [comment]
Carpet Fixes is the last mod we are waiting on to start my family's 1.21 server, we would really love if it could be updated soon

(I would go to the discord to check on progress, etc. but the link is expired)

**Comment by grillotine** [comment]
you can check the carpet discord for update info - https://discord.gg/XBRrsuJf - but from what i see theres no 1.21 planned release date

**Comment by MolodoyDeputat** [comment]
can you send another one link to Discord?

**Comment by IamKsandr707** [comment]
Any updates? I can't find any information about port to new versions :(

**Comment by Makiru-M** [comment]
still waiting

**Comment by IamKsandr707** [comment]
![image](https://github.com/user-attachments/assets/dc85ba39-de7d-4cf5-9f12-b77b2904be47)
Not enough time... So I'll try to make port of it maybe.

**Comment by Makiru-M** [comment]
> ![image](https://private-user-images.githubusercontent.com/92206294/360967808-dc85ba39-de7d-4cf5-9f12-b77b2904be47.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ1NzM4MjQsIm5iZiI6MTcyNDU3MzUyNCwicGF0aCI6Ii85MjIwNjI5NC8zNjA5Njc4MDgtZGM4NWJhMzktZGU3ZC00Y2Y1LTlmMTItYjc3YjI5MDRiZTQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI1VDA4MTIwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMxMjUyNGRlN2RhMmNhYmUxNWU0ZTI0NGU4ZDE2MDZiZjQ1ZjY3NjMzMGNjYjg1Y2RkMTZiMThkZDhhZDA5N2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xYNPkjAJh7GapTwVFXClR5GqATmD3P6vhRu4v2lspb4) Not enough time... So I'll try to make port of it maybe.

Ok. It would be nice

**Comment by lunaDHD** [comment]
> ![image](https://private-user-images.githubusercontent.com/92206294/360967808-dc85ba39-de7d-4cf5-9f12-b77b2904be47.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2MTU5MDEsIm5iZiI6MTcyNDYxNTYwMSwicGF0aCI6Ii85MjIwNjI5NC8zNjA5Njc4MDgtZGM4NWJhMzktZGU3ZC00Y2Y1LTlmMTItYjc3YjI5MDRiZTQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI1VDE5NTMyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyYmRjNDFiNzQzZjJmMTA4YmIxMTA2YTVkYWFjMWM0YzNiNDFhYWYzYTVmMzU1ODQ5OTIzMTdhYzNjOGRkZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xsY4ZdedN7LqUOYmYSQ0aqwc1psF9VdJoScBxVSTWEI) Not enough time... So I'll try to make port of it maybe.

PLEASE port it i beg of you

**Comment by Ksandr707-yt** [comment]
![image](https://github.com/user-attachments/assets/e1da83b4-3baf-446e-b874-375b92ddb557)
There is a few errors on server loading, but almost there.

**Comment by Ksandr707-yt** [comment]
![image](https://github.com/user-attachments/assets/646da874-e3a7-4822-bf3d-801bd8f16991)
Work In Progress...

**Comment by electricsteve** [comment]
Hey @Ksandr707-yt  lmk if you need help, I have a lot of spare time and can help.

**Comment by JorgeSotoOrellana** [comment]
Does anyone has the Discord link?
Please I want to get in there.

**Comment by K1miram** [comment]
@JorgeSotoOrellana, https://discord.gg/qzqUdqTcU9

**Comment by tomalbrc** [comment]
Working on an update, there is a 1.21-1.21.1 version here https://github.com/tomalbrc/carpet-fixes
Seems like some bugs were fixed by mojang when they switched to components, but not all.
There are also 2 todos for some minor stuff which is the reason I am not making a PR (at least yet until those are clarified)

**Comment by FxMorin** [comment]
@tomalbrc Looked it over.
You can remove `saferItemTransfers`, and `ResetChunksCommand` probably got removed.

**Comment by bravely-beep** [comment]
@tomalbrc Sorry to bother you, just wondering if you're still planning on working on the 1.21/1.21.1 port? It'd really come in handy for my modpack (specifically the MC-103452 fix).
